src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AspNetCore6Plus/BrowserInjectingStreamWrapper.cs

```diff
@@ -65,36 +66,43 @@ public override void SetLength(long value)
 
         public override void Write(byte[] buffer, int offset, int count)
         {
-            if (IsHtmlResponse())
-            {
-                var curBuf = buffer.AsMemory(offset, count).ToArray();
-                _agent.TryInjectBrowserScriptAsync(_context.Response.ContentType, _context.Request.Path.Value, curBuf,
-                        _baseStream)
-                    .GetAwaiter().GetResult();
-            }
-            else
+            if (!_context.Items.ContainsKey(InjectingRUM))  // pass through without modification if we're already in the middle of injecting
```

Comment:
Since this check is used in a couple of place it might be good to extract it into a method, such as `CurrentlyInjecting()`. Then the code could read like `if (!CurrentlyInjecting())`.

Reply:
Reasonable suggestion. I refactored a bit.